File: server/Models/HarvestStatusChange.cs
namespace Server.Models;

public class HarvestStatusChange
{
    public int Id { get; set; }

    public int HarvestId { get; set; }

    public HarvestStatus? FromStatus { get; set; }

    public HarvestStatus ToStatus { get; set; }

    public DateOnly Date { get; set; }

    public string? Note { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: web/src/types/harvest-status-change.ts
import type { HarvestStatus } from '@/types/harvest';

export type HarvestStatusChange = {
  id: number;
  harvestId: number;
  fromStatus: HarvestStatus | null;
  toStatus: HarvestStatus;
  date: string;
  note: string | null;
  createdAt: string;
};
